packages/apollo-server-core/src/__tests__/ApolloServerBase.test.ts

```diff
@@ -30,6 +30,53 @@ describe('ApolloServerBase construction', () => {
     ).not.toThrow();
   });
 
+  it('succeeds when passed a graphVariant in construction', () => {
+    let serverBase;
+    expect(
+      () =>
+        serverBase = new ApolloServerBase({
+          schema: buildServiceDefinition([{ typeDefs, resolvers }]).schema,
+          engine: {
+            graphVariant: 'foo',
+            apiKey: 'not:real:key',
+          },
+        }),
+    ).not.toThrow();
```

Comment:
I think we could be more specific here and test for what we _don't_ want thrown, in terms of specific error message.  Otherwise this test ends up failing when some other thing which intentionally fails this pattern crops up, _red herring during development_, etc.  We can avoid that.

Reply:
On the flip side, it means that if someone changes the error message, then the test which is looking for it will continue to pass silently, but will be doing nothing, right?

Reply:
@zionts A defensive best-of-both-worlds technique would be test for each and have the tests rely on the same error message string.

Reply:
But like... what error message are we looking for? I can think of a couple different error messages that are reasonable to test for. 

* defined both graphVariant and schemaTag
* unknown key / TypeError

Will we actually keep this up to date as we add more? Will it actually be a red herring as opposed to a good test update, when we add, e.g. "throw on malformed API key" that causes this test to error?

I would really prefer to leave this as just making sure this configuration _doesn't throw anything_ and updating the test + adding another if another behavior through a different feature causes this test to throw.